operators/src/processing/reprojection.rs

```diff
@@ -0,0 +1,399 @@
+use std::marker::PhantomData;
+
+use futures::TryStreamExt;
+use geoengine_datatypes::{
+    operations::reproject::{CoordinateProjection, CoordinateProjector, Reproject},
+    spatial_reference::SpatialReference,
+};
+use serde::{Deserialize, Serialize};
+use snafu::ensure;
+
+use crate::{
+    engine::{
+        ExecutionContext, InitializedOperator, InitializedOperatorImpl, InitializedVectorOperator,
+        Operator, QueryContext, QueryRectangle, TypedVectorQueryProcessor, VectorOperator,
+        VectorQueryProcessor, VectorResultDescriptor,
+    },
+    error,
+    util::Result,
+};
+
+use super::map_query::MapQueryProcessor;
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct ReprojectionParams {
+    pub target_spatial_reference: SpatialReference,
+}
+
+pub type Reprojection = Operator<ReprojectionParams>;
+pub type InitializedReprojection =
+    InitializedOperatorImpl<(), VectorResultDescriptor, (SpatialReference, SpatialReference)>;
+
+#[typetag::serde]
+impl VectorOperator for Reprojection {
+    fn initialize(
+        self: Box<Self>,
+        context: &dyn ExecutionContext,
+    ) -> Result<Box<InitializedVectorOperator>> {
+        ensure!(
+            self.vector_sources.len() == 1,
+            error::InvalidNumberOfVectorInputs {
+                expected: 1..2,
+                found: self.vector_sources.len()
+            }
+        );
+        ensure!(
+            self.raster_sources.is_empty(),
+            error::InvalidNumberOfRasterInputs {
+                expected: 0..1,
+                found: self.raster_sources.len()
+            }
+        );
+
+        let initialized_vector_sources = self
+            .vector_sources
+            .into_iter()
+            .map(|o| o.initialize(context))
+            .collect::<Result<Vec<Box<InitializedVectorOperator>>>>()?;
+
+        let in_desc: &VectorResultDescriptor = initialized_vector_sources[0].result_descriptor();
+        let out_desc = VectorResultDescriptor {
+            spatial_reference: self.params.target_spatial_reference.into(),
+            data_type: in_desc.data_type,
+            columns: in_desc.columns.clone(),
+        };
+
+        let state = (
+            Option::from(in_desc.spatial_reference).unwrap(),
+            self.params.target_spatial_reference,
+        );
+
+        Ok(
+            InitializedReprojection::new((), out_desc, vec![], initialized_vector_sources, state)
+                .boxed(),
+        )
+    }
+}
+
+impl InitializedOperator<VectorResultDescriptor, TypedVectorQueryProcessor>
+    for InitializedReprojection
+{
+    fn query_processor(&self) -> Result<TypedVectorQueryProcessor> {
+        let state = self.state;
+        match self.vector_sources[0].query_processor()? {
+            // TODO: use macro for that
```

Comment:
There is `call_on_generic_vector_processor!`

Reply:
i guess i can't use it since i use a different processor for Data

Reply:
Hmm, then I guess just remove the TODO.

Reply:
bb87c88